src/Type/Constant/ConstantArrayTypeBuilder.php

```diff
@@ -314,4 +322,12 @@ public function isList(): bool
 		return $this->isList->yes();
 	}
 
+	public function resetNextAutoIndexToZeroIfNegative(): void
```

Comment:
I'd thought about this method that it should be set once after creation and not before every `setOffsetValueType` call. It should change the behaviour for all the calls, not just for the next one. Instead of this logic, this method should switch a `bool` property so that the computation of `nextAutoIndexes` changes based on that.